include/Probe.php

```diff
@@ -0,0 +1,1114 @@
+
+			$lrdd = self::xrd($host);
+			if (!$lrdd)
+				return self::mail($uri, $uid);
+
+			$addr = $uri;
+		} else {
+			$parts = parse_url($uri);
+			if (!isset($parts["scheme"]) OR
+				!isset($parts["host"]) OR
+				!isset($parts["path"]))
+				return false;
+
+			// todo: Ports?
+			$host = $parts["host"];
+
+			if ($host == 'twitter.com')
+				return array("network" => NETWORK_TWITTER);
+
+			$lrdd = self::xrd($host);
+
+			$path_parts = explode("/", trim($parts["path"], "/"));
+
+			while (!$lrdd AND (sizeof($path_parts) > 1)) {
+				$host .= "/".array_shift($path_parts);
+				$lrdd = self::xrd($host);
+			}
+			if (!$lrdd)
+				return self::feed($uri);
+
+			$nick = array_pop($path_parts);
+			$addr = $nick."@".$host;
+		}
+		$webfinger = false;
+
+		/// @todo Do we need the prefix "acct:" or "acct://"?
+
+		foreach ($lrdd AS $key => $link) {
+			if ($webfinger)
+				continue;
+
+			if (!in_array($key, array("lrdd", "lrdd-xml", "lrdd-json")))
+				continue;
+
+			// Try webfinger with the address (user@domain.tld)
+			$path = str_replace('{uri}', urlencode($addr), $link);
+			$webfinger = self::webfinger($path);
+
+			// If webfinger wasn't successful then try it with the URL - possibly in the format https://...
+			if (!$webfinger AND ($uri != $addr)) {
+				$path = str_replace('{uri}', urlencode($uri), $link);
+				$webfinger = self::webfinger($path);
+
+				// Since the detection with the address wasn't successful, we delete it.
+				if ($webfinger) {
+					$nick = "";
+					$addr = "";
+				}
+			}
+
+		}
+		if (!$webfinger)
+			return self::feed($uri);
+
+		$result = false;
+
+		logger("Probing ".$uri, LOGGER_DEBUG);
+
+		if (in_array($network, array("", NETWORK_DFRN)))
+			$result = self::dfrn($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_DIASPORA))
+			$result = self::diaspora($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_OSTATUS))
+			$result = self::ostatus($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_PUMPIO))
+			$result = self::pumpio($webfinger);
+		if ((!$result AND ($network == "")) OR ($network == NETWORK_FEED))
+			$result = self::feed($uri);
+		else {
+			// We overwrite the detected nick with our try if the previois routines hadn't detected it.
+			// Additionally it is overwritten when the nickname doesn't make sense (contains spaces).
+			if ((!isset($result["nick"]) OR ($result["nick"] == "") OR (strstr($result["nick"], " "))) AND ($nick != ""))
+				$result["nick"] = $nick;
+
+			if ((!isset($result["addr"]) OR ($result["addr"] == "")) AND ($addr != ""))
+				$result["addr"] = $addr;
+		}
+
+		logger($uri." is ".$result["network"], LOGGER_DEBUG);
+
+		if (!isset($result["baseurl"]) OR ($result["baseurl"] == "")) {
+			$pos = strpos($result["url"], $host);
+			if ($pos)
+				$result["baseurl"] = substr($result["url"], 0, $pos).$host;
+		}
+
+		return $result;
+	}
+
+	/**
+	 * @brief Do a webfinger request
```

Comment:
Can you add a short description what a webfinger request is? So newbies could understand what happens